### PR TITLE
add default json when no opts are present

### LINKED ISSUE #5: add default .json for compilation
i think it'd helpful to include a default json export and when you didn't pass `.json` file to extension, it'd use that. 

this likely wouldn't account for everyone's workflow but it would let a theme developer to get started much faster. we could just make all of the information generic, similarly how tumblr does in their **Customize** view of the admin dashboard.

### PR DESCRIPTION
**Pull request by kylemac**
- add `publish` to makefile
- add `.scss` to blacklisted file ext
- cleans up coffeelint script
- process your template using a default tumblr json file when you don't pass any opts to `roots-tumblr`
- closes #5

### CODE REVIEW

File: lib/default.json [hunk 1]
```diff
@@ -0,0 +1,1390 @@
...
+  "block:StretchHeaderImage": true,
+  "block:NoStretchHeaderImage": false,
+  "HeaderImage-640": "http://assets.tumblr.com/images/default_header/optica_pattern_09_640.png?_v=abe6f565397f54e880c2b76e6fc2022e",
+  "HeaderImage-1024": "http://assets.tumblr.com/images/default_header/optica_pattern_09_1280.png?_v=abe6f565397f54e880c2b76e6fc2022e",
+  "block:IfHeaderImage": true,
+  "block:IfNotHeaderImage": false,
+  "TitleFont": "'Gibson', sans-serif",
+  "Title": "Untitled",
+  "BlogTitle": "Untitled",
+  "Name": "verycreatorwolf",
```
Comment (notslang) [inline lib/default.json]: I'm probably going to take `verycreatorwolf` down later, since I don't use tumblr

Comment (kylemac) [inline lib/default.json]: ok, i'll create a carrot one that we can use as default

File: lib/index.coffee [hunk 2]
```diff
@@ -6,13 +6,13 @@ path = require 'path'
...
 BLACKLISTED_FILE_EXTENSIONS = [
-  '.jpg', '.gif', '.png', '.css', '.styl', '.js', '.coffee'
+  '.jpg', '.gif', '.png', '.css', '.styl', '.js', '.coffee', '.scss'
 ]
 
 module.exports = (data) ->
   class TumblrCompiler
     constructor: ->
-      @data = {}
+      @data = data || require './default'
```
Comment (notslang) [inline lib/index.coffee]: could probably move this out of the constructor and just do:

``` coffee
module.exports = (data) ->
  data ?= require './default'
```

Comment (notslang) [inline lib/index.coffee]: actually, if that function wrapper isn't needed, then it should be kept in the constructor and we should just get rid of the wrapper. then we could do:

``` coffee
class TumblrCompiler
  constructor: (@data) ->
    @data ?= require './default'
```

File: test/fixtures/withData/package.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "test",
+  "dependencies": {
+    "jade": "*"
+  }
+}
```
Comment (notslang) [inline test/fixtures/withData/package.json]: ~~trailing newline~~

File: lib/index.coffee (hunk 2 above)
Comment (kylemac) [inline lib/index.coffee]: i liked:

``` coffee
module.exports = (data) ->
  class TumblrCompiler
    data ?= require './default'

    fs: ->
      category: 'tumblr-markup'
      extract: true
      detect: (f) ->
        path.extname(f.relative) not in BLACKLISTED_FILE_EXTENSIONS

    compile_hooks: ->
      category: 'tumblr-markup'
      'after_file': (ctx) ->
        # just a crude extension check... the compiler will let the markup pass
        # through unchanged if it turns out this is just regular html
        if ctx['out_ext'] is 'html'
          ctx.content = compile(ctx.content.toString('utf8'), data)
```

i didn't know you could do the `?=` . the function wrapper seemed to be necessary.

**Comment by kylemac** [comment]
@slang800 made those updates. i'll squash before merging once you give the +1 and the tests :tennis:

File: Makefile [hunk 4]
```diff
@@ -5,3 +5,8 @@ build:
 unbuild:
 	rm -rf lib
 	mv src lib
+
+publish:
```
Comment (notslang) [inline Makefile]: I'd add some sort of a check to see that it actually built correctly before publishing, also unbuild should really make sure that the src directory exists before it removes lib

**Comment by notslang** [comment]
other than the makefile & squashing, this is looking good :+1:

File: Makefile (hunk 4 above)
Comment (kylemac) [inline Makefile]: yeah, i've run into that issue in the past too. i'll remove it

**Pull request merged**